src/app/map/[id]/Tile/item.buttons.static.tsx

```diff
@@ -1,100 +1,138 @@
+"use client";
+
 import Link from "next/link";
+import { useSearchParams } from "next/navigation";
+import { Maximize2, Minimize2, Crosshair } from "lucide-react";
 import type { HexTileData } from "../State/types";
-import { Maximize2, Minimize2 } from "lucide-react";
 import type { TileScale } from "./base.static";
 
 interface TileButtonsProps {
   item: HexTileData;
   scale?: TileScale;
-  pathname: string;
-  currentSearchParamsString: string;
   allExpandedItemIds: string[];
   hasChildren: boolean;
+  isCenter?: boolean;
 }
 
-const getHref = (
+const _getButtonPositioning = (scale: number) => {
+  return "absolute left-1/2 -translate-x-1/2 top-[15%] z-30";
+};
+
+const _getExpandUrl = (
   item: HexTileData,
-  pathname: string,
-  currentSearchParamsString: string,
+  currentSearchParams: URLSearchParams,
   allExpandedItemIds: string[],
-) => {
-  const itemId = item.metadata.dbId;
-  const isExpanded = allExpandedItemIds.includes(itemId);
-
-  const newParams = new URLSearchParams(currentSearchParamsString);
-  const newExpandedItemsList = isExpanded
-    ? allExpandedItemIds.filter((id) => id !== itemId)
-    : [...allExpandedItemIds, itemId];
+): string => {
+  const isExpanded = allExpandedItemIds.includes(item.metadata.dbId);
+  const newParams = new URLSearchParams(currentSearchParams);
 
-  if (newExpandedItemsList.length > 0) {
-    newParams.set("expandedItems", newExpandedItemsList.join(","));
+  if (isExpanded) {
+    // Remove from expanded items
+    const filteredIds = allExpandedItemIds.filter(
+      (id) => id !== item.metadata.dbId,
+    );
+    if (filteredIds.length > 0) {
+      newParams.set("expandedItems", filteredIds.join(","));
+    } else {
+      newParams.delete("expandedItems");
+    }
```

Comment:
abstract this in a private helper, it then should look like this:
`
isExpanded ? collapse(newParams, item) : expand(newParams, item)
`